src/library/scala/collection/immutable/Set.scala

```diff
@@ -166,7 +193,11 @@ object Set extends IterableFactory[Set] {
       if (elem == elem1) new Set1(elem2)
       else if (elem == elem2) new Set1(elem1)
       else this
-    def iterator: Iterator[A] = (elem1 :: elem2 :: Nil).iterator
+    def iterator: Iterator[A] = new SetNIterator[A](size) {
+      def apply(i: Int) = getElem(i)
+    }
+    private def getElem(i: Int) = i match { case 0 => elem1 case 1 => elem2 }
```

Comment:
Direct access to `elemN` from the inner class would cause the fields to end up JVM public with expanded names. This breaks serialization stability of this class. I was surprised that `SetN` doesn't use `collection.generic.DefaultSerializable` to avoid such problems. @szeiger is that intentional or just an oversight?

Reply:
Intentional. JVM serialization should be more efficient and compact than our serialization proxy. All the specialized small-size collections use standard serialization.